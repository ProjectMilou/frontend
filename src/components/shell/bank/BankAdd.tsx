import React, { useState } from 'react';
import { Container } from '@material-ui/core';
import BankAccountService from '../../../services/BankAccountService';
import BankAddSearch from './BankAddSearch';
import BankAddDialog from './BankAddDialog';

export enum State {
  search,
  add,
  success,
}

const BankAdd: React.FC = () => {
  type State = 'search' | 'add' | 'success';
  const [link, setLink] = useState('');
  const [userState, setUserState] = useState<State>('search');

  const handleAdding = (id: number) => {
    setUserState('add');
    BankAccountService.add(id).then((v) => {
      setLink(v);
    });
  };

  return (
    <Container>
      {userState === 'search' && <BankAddSearch handleAdding={handleAdding} />}
      {userState === 'add' && <BankAddDialog link={link} />}
    </Container>
  );
};

export default BankAdd;
