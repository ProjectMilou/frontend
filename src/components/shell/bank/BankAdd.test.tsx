import * as React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import BankAdd from './BankAdd';
import mockResponse from './BankAddMock';

beforeEach(() => {
  fetchMock.resetMocks();
});

describe('BankSearch', () => {
  test('Search field exists', () => {
    render(<BankAdd />);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const { container } = render(<BankAdd />);
    expect(container).toMatchSnapshot();
  });

  test('should display fetch response', async () => {
    const { getByRole } = render(<BankAdd />);
    const input = getByRole('textbox') as HTMLInputElement;

    expect(input).toBeInTheDocument();
    fetchMock.mockResponseOnce(JSON.stringify(mockResponse));
    fireEvent.change(input, { target: { value: 'a' } });

    expect(window.fetch).toHaveBeenCalledWith(
      'https://api.milou.io/user/bank/search/a',
      expect.objectContaining({ method: 'GET' })
    );
    expect(window.fetch).toHaveBeenCalledTimes(1);
    await waitFor(() => {
      expect(screen.getByText(/GAD Test/i)).toBeInTheDocument();
    });
  });
});
