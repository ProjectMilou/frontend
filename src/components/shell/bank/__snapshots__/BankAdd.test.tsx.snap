// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankSearch should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        aria-label="text"
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused"
      >
        <input
          class="MuiInputBase-input MuiInput-input"
          placeholder="shell.bank.search.enter"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
